### PR TITLE
[RELEASE-1.6][SRVKS-982] Bump envoy image to 1.21 and add compatibility with 1.22 (#921)

### PR DESCRIPTION
**Pull request by nak3**
This is cherry-pick of 2137d2f5838753340a7403fec6f5966f0dd85922

/cc @skonto @maschmid

### CODE REVIEW

File: .github/workflows/kind-e2e.yaml [hunk 1]
```diff
@@ -26,8 +26,8 @@ jobs:
 
         gateway:
         - quay.io/maistra/proxyv2-ubi8:2.1.0
-        - docker.io/envoyproxy/envoy:v1.20-latest
         - docker.io/envoyproxy/envoy:v1.21-latest
+        - docker.io/envoyproxy/envoy:v1.22-latest
```
Comment (skonto) [inline .github/workflows/kind-e2e.yaml:30]: So 1.22 is 2.3.0 downstream correct?

File: config/300-gateway.yaml [hunk 2]
```diff
@@ -47,7 +47,7 @@ spec:
             - --log-level info
           command:
             - /usr/local/bin/envoy
-          image: docker.io/envoyproxy/envoy:v1.20-latest
+          image: docker.io/envoyproxy/envoy:v1.21-latest
```
Comment (skonto) [inline config/300-gateway.yaml:50]: Shouldn't this be 1.22?

Comment (nak3) [reply config/300-gateway.yaml:50]: No, not necessary. The image is not used.

**Pull request merged**